ARTstor.js

```diff
@@ -34,98 +37,1043 @@
+                        var jsonDOC = eval('(' + text + ')');
+                        var urlDOC = server + "thumb" + jsonDOC.imageUrl;
+                        artItem.attachments.push({
+                            title: "Artstor Word Attachment",
+                            url: urlDOC,
+                            mimeType: "application/msword"
+
+                        });
+                    });
+                    artItem.url = url;
+                    break;
+                case "23":
+                    var serviceURL4XXLS = serverURL + "secure/imagefpx/" + item + "/23"; //XLS
+                    Zotero.Utilities.HTTP.doGet(serviceURL4XXLS, function(text) {
+                        var jsonXLS = eval('(' + text + ')');
+                        var urlXLS = server + "thumb" + jsonXLS.imageUrl;
+                        artItem.attachments.push({
+                            title: "Artstor Excel Attachment",
+                            url: urlXLS,
+                            mimeType: "application/excel"
+
+                        });
+                    });
+                    artItem.url = url;
+                    break;
+                case "24":
+                    var serviceURL4VID = serverURL + "secure/imagefpx/" + item + "/24"; //Video
+                    Zotero.Utilities.HTTP.doGet(serviceURL4VID, function(text) {
+                        var jsonVID = eval('(' + text + ')');
+                        var urlVID = jsonVID.imageUrl;
+                        artItem.attachments.push({
+                            title: "Artstor Video File Attachment",
+                            url: urlVID,
+                            mimeType: "text/html"
+
+                        });
+                    });
+                    artItem.url = url;
+                    break;
+            }
+            var service4Commentary = serverURL + "/secure/icommentary/" + item;
+            Zotero.Utilities.HTTP.doGet(service4Commentary, function(text) {
+
+                var jsonCommentary = eval('(' + text + ')');
+                Zotero.debug("commentary " + jsonCommentary.numberOfCommentaries);
+                for (var j = 0; j < jsonCommentary.numberOfCommentaries; j = j + 1) {
+                    if (jsonCommentary.ICommentary[j].status == 2) {
+                        //public commentary
+                        var comment = "";
+
+                        if (jsonCommentary.ICommentary[j].ownerName == "") {
+                            comment = "Note: "
+
+                        } else {
+
+                            comment = "Note by: " + jsonCommentary.ICommentary[j].ownerName + " -  ";
+
+                        }
+                        comment += jsonCommentary.ICommentary[j].commentary;
+                        artItem.notes.push({
+                            note: comment
+                        });
+                    }
+
+                }
+
+
+            });
+            artItem.complete();
```

Comment:
This cannot be called synchronously. You will have to call it from "done" handlers for the `doGet` calls you use above. Since there are two of them, this gets more complicated. You'll have to make your commentaries call follow your attachments call and then at the end call `artItem.complete()`